### ISSUE TITLE
DESCRIBE KEYSPACE emits broken output with nested UDTs in collections

### ISSUE THREAD

**Issue by banjiewen** [scylladb/scylla#2494]
*Installation details*
Scylla version (or git commit hash): 1.6.0
Cluster size: 3
OS (RHEL/CentOS/Ubuntu/AWS AMI): Ubuntu

When calling `DESCRIBE KEYSPACE` on a keyspace with UDTs nested inside collections, the resulting output will not be sorted in an order that supports re-creation (e.g., for backup/restore purposes). The following example demonstrates the issue:

```
scylla@cqlsh> create keyspace test with replication = { 'class' : 'SimpleStrategy', 'replication_factor': 3 } ;
scylla@cqlsh> create type test.z ( value int );
scylla@cqlsh> create type test.a ( zs frozen<list<frozen<z>>> );
scylla@cqlsh> create table test.test (id int primary key, foobar frozen<a>);
scylla@cqlsh> describe keyspace test;
CREATE KEYSPACE test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '3'}  AND durable_writes = true;
CREATE TYPE test.a (
    zs frozen<list<frozen<z>>>
);
CREATE TYPE test.z (
    value int
);
CREATE TABLE test.test (
    id int PRIMARY KEY,
    foobar frozen<a>
) WITH bloom_filter_fp_chance = 0.01
    AND caching = '{"keys":"ALL","rows_per_partition":"ALL"}'
    AND comment = ''
    AND compaction = {'class': 'SizeTieredCompactionStrategy'}
    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 0
    AND gc_grace_seconds = 864000
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = '99.0PERCENTILE';
```

Note that without the `list` wrapper, a similar example sorts properly:

```
scylla@cqlsh> create keyspace test2 with replication = { 'class' : 'SimpleStrategy', 'replication_factor': 3 } ;
scylla@cqlsh> create type test2.z ( value int );
scylla@cqlsh> create type test2.a ( z frozen<z> );
scylla@cqlsh> create table test2.test (id int primary key, foobar frozen<a>);
scylla@cqlsh> describe keyspace test2;
CREATE KEYSPACE test2 WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '3'}  AND durable_writes = true;
CREATE TYPE test2.z (
    value int
);
CREATE TYPE test2.a (
    z frozen<z>
);
CREATE TABLE test2.test (
    id int PRIMARY KEY,
    foobar frozen<a>
) WITH bloom_filter_fp_chance = 0.01
    AND caching = '{"keys":"ALL","rows_per_partition":"ALL"}'
    AND comment = ''
    AND compaction = {'class': 'SizeTieredCompactionStrategy'}
    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 0
    AND gc_grace_seconds = 864000
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = '99.0PERCENTILE';
```

**Comment by duarten** [comment]
This is working fine for me with 1.7 and master. The logic to sort the user types is implemented in the [python driver](https://github.com/datastax/python-driver/blob/25e6cf3d0a503149481eb9d7bd2d2084526f62c9/cassandra/metadata.py#L691). 

Which version of the python driver are you using?

**Comment by banjiewen** [comment]
This is from a 1.6.0 Scylla using scylla-tools@1.6.4-20170425.9e9fbd7-ubuntu1.

**Comment by banjiewen** [comment]
Confirmed working on a 1.7.1 with scylla-tools@1.7.1-20170607.c60c113-0ubuntu1.

```
scylla@cqlsh> create keyspace test with replication = { 'class' : 'SimpleStrategy', 'replication_factor': 3 };
scylla@cqlsh> create type test.z ( value int );
scylla@cqlsh> create type test.a ( zs frozen<list<frozen<z>>> );
scylla@cqlsh> create table test.test (id int primary key, foobar frozen<a>);
scylla@cqlsh> describe keyspace test;

CREATE KEYSPACE test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '3'}  AND durable_writes = true;

CREATE TYPE test.z (
    value int
);

CREATE TYPE test.a (
    zs frozen<list<frozen<z>>>
);

CREATE TABLE test.test (
    id int PRIMARY KEY,
    foobar frozen<a>
) WITH bloom_filter_fp_chance = 0.01
    AND caching = '{"keys":"ALL","rows_per_partition":"ALL"}'
    AND comment = ''
    AND compaction = {'class': 'SizeTieredCompactionStrategy'}
    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 0
    AND gc_grace_seconds = 864000
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = '99.0PERCENTILE';

```

**Comment by duarten** [comment]
This was probably fixed by the update from version 2.7.2 of the cassandra driver to 3.5.0.

**Comment by banjiewen** [comment]
👍
